tuxemon/states/combat/combat.py

```diff
@@ -284,7 +294,18 @@ def update(self, time_delta: float) -> None:
         * update the new phase, or the current one
         """
         super().update(time_delta)
-        if not self._animation_in_progress and all(
+        self._text_animation_time_left -= time_delta
+        if (
+            self.no_ongoing_text_animation()
+            and len(self.text_animations_queue) > 0
```

Comment:
```suggestion
            and self.text_animations_queue
```

In python we generally check the truthiness of an iterable rather than calculating its length.

Reply:
You are right, it's more pythonist your way.

To be honest I hesitated about this one. I thought verifying the length was more descriptive, and I always prefer when things are more readable. Also, this would be evaluated to true in more cases than it should. Since even if `text_animations_queue` is of the wrong type, the assertion would be passing as well.

But anyway, I was considering it this way also and has you said, it's the recommended way to go.
Changing it.